internal/bminventory/inventory.go

```diff
@@ -2602,8 +2602,7 @@ func (b *bareMetalInventory) updatePlatformParams(params installer.V2UpdateClust
 }
 
 func (b *bareMetalInventory) setDiskEncryptionUsage(c *models.Cluster, diskEncryption *models.DiskEncryption, usages map[string]models.Usage) {
-
-	if c.DiskEncryption == nil {
+	if diskEncryption == nil || diskEncryption.EnableOn == nil {
```

Comment:
Maybe we can extend `setDiskEncryptionWithDefaultValues` to fit to `updateCluster` as well as `registerCluster` instead of rechecking inner fields of `DiskEncryption` ?

Seems like I forgot to set the default in `updateCluster` as well.

Reply:
As we already discussed, this seems to be a bit bigger issue than it initially seemed. However, given that fixing it properly would be a bigger change (and we need to have a backported fix to ocm-2.4), I'm not sure whether it wouldn't be better to have 2 fixes

1) as simple as possible fix, just in order to fix the error raised in the BZ
2) proper fix for handling default values in all the scenarios

All the points you raised would be perfect for (2), but then again, it would be a bigger change and backporting it creates a potential for missing something very close to the code freeze

cc @flaper87 @celebdor, as I'm still not sure what is the level of support of this feature in 2.4

Reply:
I think I would rather do this in 2 steps. Let's fix the bug first, then do the bigger change to improve the implementation.

When backporting (or in general, really :stuck_out_tongue_closed_eyes: ), the smaller the PR, the better :)

Reply:
After searching for this is issue I discovered that there is a bug in swagger that doesn't seem to be fixed and there is a workaround for it but it will complicate things in another way - once defaults work correctly, we are unable to update the cluster with DiskEncryption without overriding all the non-mentioned field in the object to the default values unless we duplicate the object in swagger.

I rather just fix the merge in that case.